cmd/cmd_volume.go

```diff
@@ -225,13 +227,130 @@ func volumeTreeCommandHandler(cmd *cobra.Command, args []string) {
 	reader.Close()
 }
 
+func volumeLsCommand() *cobra.Command {
+	var cmdLs = &cobra.Command{
+		Use:   "ls <volume_name:volume_uuid> <path>",
+		Short: "list files and directories in volume",
+		Run:   volumeLsCommandHandler,
+		Args:  cobra.MinimumNArgs(1),
+	}
+	flags := cmdLs.PersistentFlags()
+	flags.BoolP("long-format", "l", false, "use a long listing format")
+	return cmdLs
+}
+
+func volumeLsCommandHandler(cmd *cobra.Command, args []string) {
+	reader := getLocalVolumeReader(cmd, args)
+	defer reader.Close()
+	var srcPath string
+	if len(args) < 2 {
+		srcPath = "/"
+	} else if len(args) > 2 {
+		exitForCmd(cmd, "Too many arguments")
+	} else {
+		srcPath = args[1]
+	}
+	longFormat, _ := cmd.Flags().GetBool("long-format")
+	fileInfo, err := reader.Stat(srcPath)
+	if err != nil {
+		exitWithError(fmt.Sprintf("Cannot access '%s': %v", srcPath, err))
+	}
+	switch fileInfo.Mode() {
+	case os.ModeDir:
+		dirEntries, err := reader.ReadDir(srcPath)
+		if err != nil {
+			exitWithError(fmt.Sprintf("cannot read directory '%s': %v", srcPath, err))
+		}
+		for index, entry := range dirEntries {
+			if index > 0 {
+				if longFormat {
+					fmt.Println()
+				} else {
+					fmt.Print(" ")
+				}
+			}
+			switch entry.Mode() {
+			case os.ModeDir:
+				imageLsDir(entry, longFormat)
+			case os.ModeSymlink:
+				imageLsSymlink(reader, path.Join(srcPath, entry.Name()), entry, longFormat)
+			default: // regular file
+				imageLsFile(entry, longFormat)
+			}
+		}
+	case os.ModeSymlink:
+		imageLsSymlink(reader, srcPath, fileInfo, longFormat)
+	case 0: // regular file
+		imageLsFile(fileInfo, longFormat)
+	}
+	fmt.Println()
+}
+
+func volumeCopyCommand() *cobra.Command {
+	var cmdCopy = &cobra.Command{
+		Use:   "cp <volume_name:volume_uuid> <src>... <dest>",
+		Short: "copy files from volume to local filesystem",
+		Run:   volumeCopyCommandHandler,
+		Args:  cobra.MinimumNArgs(3),
+	}
+	flags := cmdCopy.PersistentFlags()
+	flags.BoolP("recursive", "r", false, "copy directories recursively")
+	flags.BoolP("dereference", "L", false, "always follow symbolic links in volume")
+	return cmdCopy
+}
+
+func volumeCopyCommandHandler(cmd *cobra.Command, args []string) {
```

Comment:
Likewise, I would make a more generic `imageCp(flags *pflag.FlagSet, args []string, reader *fs.Reader)` function and use it in both imageCopyCommandHandler() and volumeCopyCommandHandler().